cmd/telemetrygen/internal/logs/worker.go

```diff
@@ -10,78 +10,95 @@ import (
+	index          int             // worker index
+	traceID        string          // traceID string
+	spanID         string          // spanID string
 }
 
-func (w worker) simulateLogs(res *resource.Resource, exporter exporter, telemetryAttributes []attribute.KeyValue) {
+func (w worker) simulateLogs(res *resource.Resource, exporterFunc func() (sdklog.Exporter, error), telemetryAttributes []attribute.KeyValue) {
 	limiter := rate.NewLimiter(w.limitPerSecond, 1)
 	var i int64
 
-	for w.running.Load() {
-		logs := plog.NewLogs()
-		nRes := logs.ResourceLogs().AppendEmpty().Resource()
-		attrs := res.Attributes()
-		for _, attr := range attrs {
-			nRes.Attributes().PutStr(string(attr.Key), attr.Value.AsString())
-		}
+	exporter, err := exporterFunc()
+	if err != nil {
+		w.logger.Error("failed to create the exporter", zap.Error(err))
+		return
+	}
 
-		log := logs.ResourceLogs().At(0).ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
-		log.Body().SetStr(w.body)
-		log.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
-		log.SetDroppedAttributesCount(1)
-		log.SetSeverityNumber(w.severityNumber)
-		log.SetSeverityText(w.severityText)
-		log.Attributes()
-		lattrs := log.Attributes()
-		lattrs.PutStr("app", "server")
+	defer func() {
+		w.logger.Info("stopping the exporter")
+		if tempError := exporter.Shutdown(context.Background()); tempError != nil {
+			w.logger.Error("failed to stop the exporter", zap.Error(tempError))
+		}
+	}()
 
-		if w.traceID != "" {
-			// we checked this for errors in the Validate function
-			// nolint: errcheck
-			b, _ := hex.DecodeString(w.traceID)
+	for w.running.Load() {
 
-			tid := pcommon.TraceID(b)
-			log.SetTraceID(tid)
-		}
+		var tid trace.TraceID
+		var sid trace.SpanID
 
 		if w.spanID != "" {
 			// we checked this for errors in the Validate function
 			// nolint: errcheck
 			b, _ := hex.DecodeString(w.spanID)
-
-			sid := pcommon.SpanID(b)
-			log.SetSpanID(sid)
+			sid = trace.SpanID(b)
+		}
+		if w.traceID != "" {
+			// we checked this for errors in the Validate function
+			// nolint: errcheck
+			b, _ := hex.DecodeString(w.traceID)
+			tid = trace.TraceID(b)
 		}
 
+		attrs := []log.KeyValue{log.String("app", "server")}
 		for i, attr := range telemetryAttributes {
-			lattrs.PutStr(string(attr.Key), telemetryAttributes[i].Value.AsString())
+			attrs = append(attrs, log.String(string(attr.Key), telemetryAttributes[i].Value.AsString()))
+		}
+
+		rf := logtest.RecordFactory{
+			Timestamp:         time.Now(),
+			Severity:          w.severityNumber,
+			SeverityText:      w.severityText,
+			Body:              log.StringValue(w.body),
+			Attributes:        attrs,
+			TraceID:           tid,
+			SpanID:            sid,
+			Resource:          res,
+			DroppedAttributes: 1,
+			// ObservedTimestamp time.Time
+			// TraceFlags        trace.TraceFlags
+			// InstrumentationScope *instrumentation.Scope
+			// AttributeValueLengthLimit int
+			// AttributeCountLimit       int
```

Comment:
Is there a reason to leave these lines commented out or can we delete?

Reply:
I was leaving these here are fields we may want to add details about in the future, but they can likely be removed at this point